templates/ray-data-llm/README.md

```diff
@@ -0,0 +1,162 @@
+
+
+
+```python
+import ray 
+import datasets
+
+# Load the dataset from Hugging Face into Ray Data. If you're using your own dataset,
+# refer to Ray Data APIs https://docs.ray.io/en/latest/data/api/input_output.html to load it.
+# For example, you can use ray.data.read_json(dataset_file) to load dataset in JSONL.
+
+df = datasets.load_dataset("cnn_dailymail", "3.0.0")
+ds = ray.data.from_huggingface(df["train"])
+
+```
+
+## Step 2: Define the processor config for vLLM engine
+
+We will also need to define the model configs for the LLM engine, which configures the model and compute resources needed for inference. 
+
+Some models will require you to input your [Hugging Face user access token](https://huggingface.co/docs/hub/en/security-tokens). This will be used to authenticate/download the model and **is required for official LLaMA, Mistral, and Gemma models**. You can use one of the other models which don't require a token if you don't have access to this model (for example, `neuralmagic/Meta-Llama-3.1-8B-Instruct-FP8`).
+
+Run the following cell to start the authentication flow. A VS Code overlay will appear and prompt you to enter your Hugging Face token if your selected model requires authentication. If you are using a model that does not require a token, you can skip this step. For this example, we will be using the `meta-llama/Meta-Llama-3.1-8B-Instruct` model, which requires a token.
+
+
+
+```python
+# Prompts the user for Hugging Face token if required by the model.
+from util.utils import prompt_for_hugging_face_token
+HF_TOKEN = prompt_for_hugging_face_token("meta-llama/Meta-Llama-3.1-8B-Instruct")
+```
+
+In this example, we will be using the `meta-llama/Meta-Llama-3.1-8B-Instruct` model.
+We will also need to define a configuration associated with the model we want to use to configure the compute resources, engine arguments and other inference engine specific parameters. For more details on the the model configs, see the [API doc](https://docs.anyscale.com/llms/serving/guides/bring_any_model/) on bringing your own models.
+
+
+```python
+from ray.data.llm import vLLMEngineProcessorConfig
+from util.utils import is_on_gcp_cloud
+
+# There's no a10g on GCP.
+accelerator_type = "L4" if is_on_gcp_cloud() else "A10G"
+
+processor_config = vLLMEngineProcessorConfig(
+    model_source="meta-llama/Meta-Llama-3.1-8B-Instruct",
+    engine_kwargs=dict(
+        tensor_parallel_size=1,
+        pipeline_parallel_size=1,
+        max_model_len=16384,
+        enable_chunked_prefill=True,
+        max_num_batched_tokens=2048,
+    ),
+    # Override Ray's runtime env to include the Hugging Face token. Ray is being used under the hood to orchestrate the inference pipeline.
+    runtime_env=dict(
+        env_vars=dict(
+            HF_TOKEN=HF_TOKEN,
+        ),
+    ),
```

Comment:
Echo the previous comment, we could just say "you need to put your HF token here if the model you're using is gated".